src/functions.php

```diff
@@ -758,8 +758,14 @@ function _parse_message($message)
         throw new \InvalidArgumentException('Invalid message');
     }
 
+    $message = ltrim($message, "\r\n");
+    $rawHeaders = $message;
+
     $headerDelimiterPosition = strpos($message, "\r\n\r\n");
-    $rawHeaders = substr($message, 0, $headerDelimiterPosition + 2); // We preserve the last \r\n, hence +2
+    if ($headerDelimiterPosition !== false) {
+        $rawHeaders = substr($message, 0, $headerDelimiterPosition + 2); // We preserve the last \r\n, hence +2
+    }
```

Comment:
If `$headerDelimiterPosition` is `false`, there's no valid message and I guess an exception should be thrown.